gulpfile.js

```diff
@@ -106,119 +78,27 @@ var checkCommands = function(yargs, argv, numRequired) {
-        dependency.bundle.css.forEach(function(cssFiles) {
-          cssFilePaths.push(path.join(
-            'third_party', 'static', dependencyDir, cssFiles));
-        });
-      }
-      if (dependency.bundle.hasOwnProperty('js')) {
-        dependency.bundle.js.forEach(function(jsFiles) {
-          jsFilePaths.push(path.join(
-            'third_party', 'static', dependencyDir, jsFiles));
-        });
-      }
-      if (dependency.bundle.hasOwnProperty('fontsPath')) {
-        var fontPrefix = '*.{eot,woff2,ttf,woff,eof,svg}';
-        fontFolderPaths.push(path.join('third_party', 'static', dependencyDir,
-          dependency.bundle.fontsPath, fontPrefix));
-      }
-    }
-  }
-});
-
-gulp.task('generateCss', function() {
-  requireFilesExist(cssFilePaths);
-  gulp.src(cssFilePaths)
-    .pipe(isMinificationNeeded ? sourcemaps.init() : gulpUtil.noop())
-    .pipe(isMinificationNeeded ? concat('third_party.min.css') :
-        concat('third_party.css'))
-    .pipe(isMinificationNeeded ? cleanCss({}) : gulpUtil.noop())
-    .pipe(isMinificationNeeded ? sourcemaps.write('.') : gulpUtil.noop())
-    .pipe(gulp.dest(generatedCssTargetDir));
-});
-
-gulp.task('generateJs', function() {
-  requireFilesExist(jsFilePaths);
-  gulp.src(jsFilePaths)
-    .pipe(isMinificationNeeded ? sourcemaps.init() : gulpUtil.noop())
-    .pipe(isMinificationNeeded ? concat('third_party.min.js') :
-        concat('third_party.js'))
-    .pipe(isMinificationNeeded ? uglify() : gulpUtil.noop())
-    // This maps a combined/minified file back to an unbuilt state by holding
-    // information about original files. When you query a certain line and
-    // column number in your generated JavaScript, you can do a lookup in the
-    // source map which returns the original location.
-    // http://www.html5rocks.com/en/tutorials/developertools/sourcemaps/
-    .pipe(isMinificationNeeded ? sourcemaps.write('.') : gulpUtil.noop())
-    .pipe(gulp.dest(generatedJsTargetDir));
-});
-// This task is used to copy all fonts which are used by
-// Bootstrap and font-Awesome to one folder
-var generatedFontsTargetDir = path.join(generatedTargetDir, 'fonts');
-gulp.task('copyFonts', function() {
-  gulp.src(fontFolderPaths)
-    .pipe(gulp.dest(path.join(generatedFontsTargetDir)));
-});
-
 gulp.task('gulpStartGae', function() {
   gulp.src('app.yaml')
     .pipe(gulpStartGae(gaeDevserverPath, [], params));
 });
 
-// This takes all functions  that are required for the build
-// e.g css, Js and Images
-gulp.task('build', [
-  'collectDependencyFilepaths', 'generateCss', 'copyFonts', 'generateJs']);
+gulp.task('build', shell.task(['$PYTHON_CMD scripts/build.py']));
 
 gulp.slurped = false;
 gulp.task('watch', function() {
   if (!gulp.slurped) {
     gulp.watch('gulpfile.js', ['build']);
-    gulp.watch(cssFilePaths, ['generateCss']);
-    gulp.watch(jsFilePaths, ['generateJs']);
+    gulp.watch(path.join('third_party', 'static', '**', '*.css'), ['build']);
```

Comment:
Consider making path.join(...) into a named variable -- thirdPartyCssFilepaths or similar -- so that it's easy to modify if we ever do something different there. Ditto in the next line.

Reply:
Done.